lib/game/client/game-engine.js

```diff
@@ -91,26 +91,49 @@ class GameEngine extends Game
 
     targetDisplay(targetName, target)
     {
-        // @TODO - BETA - Refactor.
+        let targetDisplayContent = targetName;
+        if (GameConst.TYPE_PLAYER === target.type) {
+            targetDisplayContent += this.generateTargetPlayedTime(target);
+        }
+        return targetDisplayContent;
+    }
+
+    generateTargetPlayedTime(target)
+    {
+        let playerTimeText = '';
         let gameManager = this.uiScene.gameManager;
-        let showPlayedTime = gameManager.config.get('client/players/playedTime/show');
-        if(0 === showPlayedTime || GameConst.TYPE_PLAYER !== target.type){
-            return targetName;
+        let showPlayedTimeConfig = gameManager.config.getWithoutLogs('client/players/playedTime/show',
+            GameConst.SHOW_PLAYER_TIME.ONLY_OWN_PLAYER);
+        if (GameConst.SHOW_PLAYER_TIME.NONE === showPlayedTimeConfig) {
+            return playerTimeText;
         }
-        let currentPlayer = gameManager.getCurrentPlayer();
-        let timeText = '';
-        let label = gameManager.config.get('client/players/playedTime/label');
-        if(0 < showPlayedTime && currentPlayer.playerId === target.id){
-            let element = gameManager.gameDom.createElement('p');
-            element.innerHTML = label+(currentPlayer.playedTime / 60 / 60).toFixed(1)+'hs';
-            timeText = element.outerHTML;
+        let currentPlayer = this.uiScene.gameManager.getCurrentPlayer();
+        if (GameConst.SHOW_PLAYER_TIME.ALL_PLAYERS === showPlayedTimeConfig ||
+            currentPlayer.playerId === target.id) {
+            let targetPlayedTime = this.obtainPlayedTime(target, currentPlayer);
+            playerTimeText += this.createPlayedTimeLabel(gameManager, targetPlayedTime);
         }
-        if(2 === showPlayedTime && sc.hasOwn(currentPlayer.players, target.id)){
-            let element = gameManager.gameDom.createElement('p');
-            element.innerHTML = label+(currentPlayer.players[target.id].playedTime / 60 / 60).toFixed(1)+'hs';
-            timeText = element.outerHTML;
+        return playerTimeText;
+    }
+
+    createPlayedTimeLabel(gameManager, playedTime)
+    {
+        let label = gameManager.config.get('client/players/playedTime/label').replace(
+            '%playedTimeInHs',
+            playedTime
+        );
+        let htmlElement = gameManager.gameDom.createElement('p');
+        htmlElement.innerHTML = label;
+        return htmlElement.outerHTML;
+    }
+
+    obtainPlayedTime(target, currentPlayer)
+    {
+        if (currentPlayer.playerId === target.id) {
```

Comment:
never use `else`, better to assing the player into a variable like:
```
let player = currentPlayer.playerId === target.id ? currentPlayer : currentPlayer.players[target.id];
// and then:
return (player.playedTime / 60 / 60).toFixed(1);
```
sometimes you could even use a single line like:
```
return ((currentPlayer.playerId === target.id ? currentPlayer : currentPlayer.players[target.id]).playedTime / 60 / 60).toFixed(1);
```
and in this case I think this would also be valid:
```
return (currentPlayer.players[target.id].playedTime / 60 / 60).toFixed(1);
```
as far I can recall currentPlayer.players contains the current player as well

Reply:
Changed!